prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,205 @@
+"""Module to define tasks for interacting with Kubernetes jobs."""
+
+from typing import Dict, Optional
+
+from kubernetes import client
+from kubernetes.client.models import V1Job, V1JobList, V1Status
+from prefect import task
+
+from prefect_kubernetes.credentials import KubernetesCredentials
+
+
+@task
+async def create_namespaced_job(
+    body: Dict,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for creating a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representing a Kubernetes V1Job specification.
+        namespace: The Kubernetes namespace to create this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
+
+    return api_client.create_namespaced_job(
+        namespace=namespace, body=body, **kube_kwargs
+    )
+
+
+@task
+async def delete_namespaced_job(
+    job_name: str,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+    delete_option_kwargs: Optional[Dict] = None,
+) -> V1Status:
+    """Task for deleting a namespaced Kubernetes job.
+
+    Args:
+        job_name: The name of a job to delete.
+        namespace: The Kubernetes namespace to delete this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+        - delete_option_kwargs: Optional keyword arguments to pass to
+            the V1DeleteOptions object (e.g. {"propagation_policy": "...",
+            "grace_period_seconds": "..."}.
+
+    Returns:
+        A Kubernetes `V1Status` object.
+    """
+
+    api_client = kubernetes_credentials.get_batch_client()
+
+    kwargs = kube_kwargs or {}
+
+    if delete_option_kwargs:
+        kwargs.update(body=client.V1DeleteOptions(**delete_option_kwargs))
+
+    return api_client.delete_namespaced_job(
+        name=job_name, namespace=namespace, **kwargs
+    )
+
+
+@task
+async def list_namespaced_job(
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
```

Comment:
I wonder if we should have `Dict[str, Any]` for all these `Dict` as part of our style guide?